Source/VCExpr/TypeErasure.cs

```diff
@@ -685,7 +685,7 @@ private struct TypeCastSet {
     [Pure]
     public override bool UnchangedType(Type type) {
       //Contract.Requires(type != null);
-      return type.IsInt || type.IsReal || type.IsBool || type.IsBv || type.IsFloat || type.IsRMode || type.IsString || type.IsRegEx || (type.IsMap && CommandLineOptions.Clo.MonomorphicArrays);
+      return type.IsInt || type.IsReal || type.IsBool || type.IsBv || type.IsFloat || type.IsRMode || type.IsString || type.IsRegEx;
```

Comment:
I have a similar question as above for the changes in this file.

Reply:
There is an invariant now that no AxiomBuilders get created/invoked for monomorphic programs.